Expensify/App
# Support validated links coming from e.com

Please review!

CC: @cead22, @tgolen, @marcaaron, @Luke9389 
### Details


### Fixed Issues
https://github.com/Expensify/Expensify/issues/177739

### Tests
See https://github.com/Expensify/Web-Expensify/pull/31977

### QA Steps
1. Log into an account on NewDot Expensify, create a Workspace if you don't already have one using the `+` on the bottom right.
2. Log out of NewDot and then log into that same account on OldDot and go to Settings -> Policy and find your Workspace there. It should have a card icon like these:
![image](https://user-images.githubusercontent.com/19364431/134262812-35531673-c2f0-48ab-9dbb-f58cced40371.png)
3. Click on its name and verify it opens a new tab, automatically logs you in, and brings you to the workspace settings page:
![image](https://user-images.githubusercontent.com/19364431/134262925-c4fd8487-f084-4ffc-98fb-2ef25bfa6c3d.png)


### Tested On


### Screenshots
Included above
